sql/2020/14_PWA/top_500_manifest_properties.sql

```diff
@@ -0,0 +1,38 @@
+#standardSQL
+# Top 500 manifest properties - based on 2019/14_04.sql
+CREATE TEMPORARY FUNCTION getManifestProps(manifest STRING)
+RETURNS ARRAY<STRING> LANGUAGE js AS '''
+try {
+  return Object.keys(JSON.parse(manifest));
+} catch (e) {
+  return null;
+}
+''';
+
+SELECT
+  client,
+  property,
+  COUNT(DISTINCT page) AS freq,
+  total,
+  COUNT(DISTINCT page) / total AS pct
+FROM (
+  SELECT
+    client,
+    page,
+    getManifestProps(body) AS properties,
+    COUNT(DISTINCT page) OVER (PARTITION BY client) AS total
+  FROM
+    `httparchive.almanac.manifests`
+  WHERE
+    date = '2020-08-01'
+  ),
+  UNNEST(properties) AS property
+GROUP BY
+  client,
+  total,
+  property,
+  client
+ORDER BY
+  freq / total DESC
+LIMIT
+  500
```

Comment:
Even top 500 properties start to get into really esoteric stuff. Perhaps restrict by frequency?

More of a pro tip, not necessary since you've already run these results.

```suggestion
HAVING
  freq >= 100
ORDER BY
  freq / total DESC
```

Reply:
Yeah this one comes straight from last year. I prefer your one as fair cut off. With the 500 arbitrary cut off some stats with same frequency will be cut off.

Will change to below, and rename it. This gives 395 results (so comparable to last year) but resolves my "fairness" issue.

```
HAVING
  freq > 10
ORDER BY
  freq / total DESC,
  property,
  client
```